sbpy/photometry/core.py

```diff
@@ -99,24 +104,306 @@ def mag2ref(mag, radius, M_sun=None):
+    >>> print(S)
+    10.0 % / 100 nm
+
+    >>> from sbpy.units import VEGAmag
+    >>> bp = ('johnson_v', 'cousins_r')
+    >>> VmR = 15.8 * VEGAmag - 15.3 * VEGAmag
+    >>> VmR_sun = 0.37 * u.mag
+    >>> S = SpectralGradient.from_color(bp, VmR - VmR_sun)
+    ...                              # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+    >>> print(S)                     # doctest: +REMOTE_DATA +FLOAT_CMP
+    12.29185986266534 % / 100 nm
+
+
+    References
+    ----------
+    .. [ADT84] A'Hearn, Dwek & Tokunaga 1984. Infrared Photometry of
+        Comet Bowell and Other Comets. ApJ 282, 803-806.
+
+    """
+
+    _equivalent_unit = u.meter**-1
+    _include_easy_conversion_members = False
+
+    def __new__(cls, value, unit=None, wave=None, wave0=None,
+                dtype=None, copy=None):
+        S = super().__new__(cls, value, unit=unit, dtype=dtype,
+                            copy=copy)
+
+        if wave is not None:
+            if np.size(wave) != 2:
+                raise ValueError(
+                    'Two wavelengths must be provided, got {}'
+                    .format(np.size(wave)))
+            S.wave = S._eff_wave(wave)
+
+        if wave0 is None and wave is not None:
+            S.wave0 = S.wave.mean()
+        else:
+            S.wave0 = wave0
+
+        return S
+
+    @classmethod
+    def _eff_wave(cls, wfb):
+        """Wavelength/frequency/bandpass to wavelength.
+
+        Bandpass is converted to effective wavelength using a solar
+        spectrum.
+
+        """
+
+        eff_wave = (0, 0) * u.um
+        sun = Sun.from_default()
+        for i in range(2):
+            if isinstance(wfb, u.Quantity):
+                eff_wave[i] = wfb[i].to(u.um)
+            else:
+                eff_wave[i] = sun.filt(wfb[i])[0]
+                log.info('Using λ_eff = {}'.format(eff_wave[i]))
+
+        return eff_wave
+
+    @classmethod
+    def from_color(cls, wfb, color):
+        r"""Initialize from observed color.
+
+
+        Parameters
+        ----------
+        wfb : two-element `~astropy.units.Quantity` or tuple
+            Wavelengths, frequencies, or bandpasses of the
+            measurement.  If a bandpass, the effective wavelength of a
+            solar spectrum will be used.  Bandpasses may be a string
+            (name) or `~synphot.SpectralElement` (see
+            :func:`~sbpy.spectroscopy.sun.Sun.filt`).
+
+        color : `~astropy.units.Quantity`, optional
+            Observed color, ``blue - red`` for magnitudes, ``red
+            / blue`` for linear units.  Must be dimensionless and have
+            the solar color removed.
+
+
+        Notes
+        -----
+
+        Computes spectral gradient from ``color_index``.
+        ``wfb[0]`` is the blue-ward of the two measurements
+
+        .. math::
+
+           S &= \frac{R(λ1) - R(λ0)}{R(λ1) + R(λ0)} \frac{2}{Δλ} \\
+           &= \frac{α - 1}{α + 1} \frac{2}{Δλ}
+
+        where R(λ) is the reflectivity, and:
```

Comment:
reflectivity or reflectance?  Have to admit that I don't really know the difference between these two words.

Reply:
I used reflectivity here since that is what Mike A. used in his paper.  I think they are interchangable, though.